src/main/java/com/google/sps/placeGuideWithCreatorPair/PlaceGuideWithCreatorPair.java

```diff
@@ -0,0 +1,51 @@
+// Copyright 2020 Google LLC
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     https://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.placeGuideWithCreatorPair;
+
+import com.google.sps.data.RepositoryType;
+import com.google.sps.placeGuide.PlaceGuide;
+import com.google.sps.user.User;
+import com.google.sps.user.repository.UserRepository;
+import com.google.sps.user.repository.UserRepositoryFactory;
+
+/**
+ * Class which unites a place guide's information with its creators information. This class is used
+ * to return the complete information about a place guide from the servlets, including the user's
+ * data, in one object.
+ */
+public class PlaceGuideWithCreatorPair {
+  private User creator;
+  private PlaceGuide placeGuide;
+  private static final UserRepository userRepository =
+      UserRepositoryFactory.getUserRepository(RepositoryType.DATASTORE);;
+
+  private PlaceGuideWithCreatorPair(User creator, PlaceGuide placeGuide) {
```

Comment:
why not :
```
public PlaceGuideWithCreatorPair(PlaceGuide placeGuide) {
  creator = userRepository.getUser(placeGuide.getCreatorId);
  this.placeGuide = placeGuide;
}
```
so we don't need the below static method?

Reply:
I prefer static factory methods in case that I don't only initialize the member variables of the class with the parameters transmitted through the constructor, because static factory methods can have a customized name suggesting what else is happening in the background. 
You are right that createPlaceGuideWithCreatorPair was not suggestive enough, so I renamed it to matchPlaceGuideWithCreator.